### PR TITLE
Fix authorization subquery variable name clash

### LINKED ISSUE #4170: @auth in combination with @populatedBy in certain instances produces errors.
### Type definitions

```GraphQL
type JWT @jwt {
    id: String
}
type User
    @authorization(
        validate: [
            { where: { node: { userId: "$jwt.id" } }, operations: [READ] }
        ]
    ) {
    userId: String! @unique
    adminAccess: [Tenant!]! @relationship(type: "ADMIN_IN", direction: OUT)
}

type Tenant
    @authorization(
        validate: [{ where: { node: { admins: { userId: "$jwt.id" } } } }]
    ) {
    id: ID! @id
    admins: [User!]! @relationship(type: "ADMIN_IN", direction: IN)
    settings: Settings @relationship(type: "VEHICLECARD_OWNER", direction: IN)
}

type Settings
    @authorization(
        validate: [
            { where: { node: { tenant: { admins: { userId: "$jwt.id" } } } } }
        ]
    ) {
    id: ID! @id
    openingDays: [OpeningDay!]!
        @relationship(type: "VALID_GARAGES", direction: OUT)
    tenant: Tenant! @relationship(type: "VEHICLECARD_OWNER", direction: OUT) # <---  this line
}

type OpeningDay
    @authorization(
        validate: [
            {
                where: {
                    node: {
                        settings: { tenant: { admins: { userId: "$jwt.id" } } }
                    }
                }
            }
        ]
    ) {
    id: ID! @id
    settings: Settings @relationship(type: "VALID_GARAGES", direction: IN)
    open: [OpeningHoursInterval!]!
        @relationship(type: "HAS_OPEN_INTERVALS", direction: OUT)
}
type OpeningHoursInterval
    @authorization(
        validate: [
            {
                where: {
                    node: {
                        openingDay: {
                            settings: {
                                tenant: { admins: { userId: "$jwt.id" } }
                            }
                        }
                    }
                }
            }
        ]
    ) {
    name: String
    openingDay: OpeningDay!
        @relationship(type: "HAS_OPEN_INTERVALS", direction: IN)
    updatedBy: String
        @populatedBy(
            callback: "getUserIDFromContext"
            operations: [CREATE, UPDATE]
        )
}
```

### Test data

_No response_

### Steps to reproduce

Clone and run test in the  following repo https://github.com/afreakk/n4j-issue-populated-by would be easiest.

Otherwise.
1. use type definition above
2. Have features like so:
```
features: {
            populatedBy: {
                callbacks: {
                    getUserIDFromContext: () => 'hi',
                },
            },
        }
```
run the following
```
apolloServer.executeOperation(
            {
                query: gql`
                    mutation addTenant($input: [TenantCreateInput!]!) {
                        createTenants(input: $input) {
                            tenants {
                                id
                                admins {
                                    userId
                                }
                            }
                        }
                    }
                `,
                variables: {
                    input: {
                        admins: {
                            create: {
                                node: { userId: myUserId },
                            },
                        },
                        settings: {
                            create: {
                                node: {
                                    openingDays: {
                                        create: {
                                            node: {
                                                open: {
                                                    create: {
                                                        node: {
                                                            name: 'lambo',
                                                        },
                                                    },
                                                },
                                            },
                                        },
                                    },
                                },
                            },
                        },
                    },
                },
            },
            { contextValue: { jwt: { id: myUserId } } }
        )
        ```

### What happened

```
+               "Neo4jError: Variable `authorization_0_0_var0` already declared in outer scope (line 81, column 50 (offset: 4407))",
+               "\"    RETURN count(authorization_0_0_this1) = 1 AS authorization_0_0_var0\"",
```

### Expected behaviour

Should work

### Version

4.3.4

### Database version

4.4.14

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by a-alle**
# Description

Fixes authorization subquery variable name clash

## Complexity

Complexity: Low

# Issue

Closes #4170 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-4195.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 38 | 52 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 79 | 95 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 49 | 59 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3286 | 3141 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2819 | 3046 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 74 | 46 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 108 | 55 | 2174 |
|  aggregations.DeeplyNestedCount | 11062443 | 11062443 | 4347 | 4372 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 154 | 186 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 76 | 90 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 244 | 207 | 3003 |
|  connections.Connection | 14082 | 14082 | 75 | 147 | 2174 |
|  connections.NestedConnection | 41118 | 41118 | 128 | 145 | 4516 |
|  create.SimpleMutation | 7 | 7 | 59 | 73 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 32 | 32 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 241 | 241 | 1040 |
|  delete.NestedDeleteInUpdate | 19198 | 19198 | 176 | 309 | 1179 |
|  2871.NestedRelationshipFilter | 23158 | 23131 | 126 | 134 | 4448 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 45 | 68 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 87 | 152 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 42 | 50 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 81 | 75 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 25 | 34 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 44 | 55 | 2174 |
|  query.QueryWhere | 9698 | 9698 | 53 | 43 | 2163 |
|  query.SimpleQueryWithNestedWhere | 9876 | 9876 | 53 | 69 | 2163 |
|  query.Nested | 10096041 | 10096041 | 10249 | 10162 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 9751 | 9822 | 2004000 |
|  query.OrFilterOnRelationships | 42268 | 42873 | 183 | 250 | 1720 |
|  query.OrFilterOnRelationshipsAndNested | 36615 | 35763 | 220 | 296 | 1720 |
|  query.QueryWithNestedIn | 13290 | 14663 | 61 | 70 | 1179 |
|  query.NestedConnectionWhere | 9834 | 9834 | 61 | 75 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10029 | 9888 | 121 | 110 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19278 | 19128 | 162 | 201 | 1646 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 187 | 223 | 1134 |
|  query.Fulltext | 80 | 80 | 34 | 38 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 46 | 51 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 14092 | 14092 | 68 | 55 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 14092 | 14092 | 68 | 74 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 14220 | 14204 | 68 | 101 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 14220 | 14204 | 91 | 126 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 14627 | 14888 | 119 | 296 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 14412 | 14566 | 111 | 159 | 2174 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 70 | 88 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14499 | 14513 | 96 | 113 | 2174 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 49 | 56 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 98 | 149 | 4516 |
|  sorting.ConnectionWithSort | 3284 | 3284 | 101 | 80 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 86 | 78 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 82 | 80 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 367 | 340 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 286 | 335 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 106 | 121 | 2002 |

</details>
        
Old Schema Generation: 38.218s
Schema Generation: 38.069s
Old Subgraph Schema Generation: 50.193s
Subgraph Schema Generation: 48.400s

File: packages/graphql/src/translate/create-create-and-params.ts [hunk 1]
```diff
@@ -157,7 +157,7 @@ function createCreateAndParams({
                             withVars: [...withVars, nodeName],
                             includeRelationshipValidation: false,
                             topLevelNodeVariable,
-                            idx: idx + index,
```
Comment (angrykoala) [inline packages/graphql/src/translate/create-create-and-params.ts:160]: Really? :man_facepalming:

File: packages/graphql/tests/integration/issues/4170.int.test.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+describe("https://github.com/neo4j/graphql/issues/4170", () => {
+    let driver: Driver;
+    let neo4j: Neo4j;
+
+    const typeDefs = gql`
+        type JWT @jwt {
+            id: String
+            roles: [String]
+        }
+        type User @authorization(validate: [{ where: { node: { userId: "$jwt.id" } }, operations: [READ] }]) {
```
Comment (angrykoala) [inline packages/graphql/tests/integration/issues/4170.int.test.ts:35]: These should use `UniqueType` to avoid flaky tests :)

**Review by angrykoala — CHANGES_REQUESTED** [review]

File: packages/graphql/tests/integration/issues/4170.int.test.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    `;
+
+    const ADD_TENANT = `
+        mutation addTenant($input: [${Tenant.name}CreateInput!]!) {
+            ${Tenant.operations.create}(input: $input) {
+                ${Tenant.plural} {
+                    id
+                    admins {
+                        userId
+                    }
```
Comment (afreakk) [inline packages/graphql/tests/integration/issues/4170.int.test.ts:88]: If you also  query
```
                                settings {
                                    openingDays {
                                        open {
                                            name
                                        }
                                    }
                                }
```
here, you will get error `Neo4jError: Variable `create_var9` not defined (line 108, column 80 (offset: 6513))`
Just noticed  this in my project, when I tried applying this `idx + 1` fix, should maybe be a different issue if the cause is very different.

Comment (a-alle) [reply packages/graphql/tests/integration/issues/4170.int.test.ts:88]: hi @afreakk, yes it is definitely a different issue but I will look into that as well. Thanks for flagging it!

**Review by angrykoala — APPROVED** [review]

**Pull request merged**